File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithFactories8.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.application.graph.TypeRef;
import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;

@KoraApp
public interface AppWithFactories8 {
    @Root
    default Object mock(GenericInterface<Integer, String> object) {
        return new Object();
    }

    default <T extends Comparable<T>> GenericImpl<T> impl(TypeRef<T> t) {
        return new GenericImpl<>();
    }


    interface GenericInterface<T1, T2> {}

    class GenericImpl<T> implements GenericInterface<T, String> {}
}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithFactories8.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.application.graph.TypeRef
import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root

@KoraApp
interface AppWithFactories8 {
    @Root
    fun mock(`object`: GenericInterface<Int, String>) = Any()

    fun <T : Comparable<T>> impl(t: TypeRef<T>): GenericImpl<T> {
        return GenericImpl()
    }

    interface GenericInterface<T1, T2>
    class GenericImpl<T> : GenericInterface<T, String>
}
